### PR TITLE
chore(package): Prettier integration

### PR DESCRIPTION
**Pull request by thetric**
As suggested by @layershifter in https://github.com/Semantic-Org/Semantic-UI-React/pull/2596#pullrequestreview-100998473 I have added `prettier` as a code formatter.

The [prettier docs](https://prettier.io/docs/en/options.html#print-width) recommends _against_  a max line length > 80, but this code base uses a print width of 120.

> **For readability we recommend against using more than 80 characters:**
>
>In code styleguides, maximum line length rules are often set to 100 or 120. However, when humans write code, they don't strive to reach the maximum number of columns on every line. Developers often use whitespace to break up long lines for readability. In practice, the average line length often ends up well below the maximum.
>
>Prettier, on the other hand, strives to fit the most code into every line. With the print width set to 120, prettier may produce overly compact, or otherwise undesirable code.

### CODE REVIEW

**Comment by layershifter** [comment]
Let's go with single quotes on JSX, it's a part of our codestyle and it will allow to reduce changes.

**Comment by thetric** [comment]
Prettier doesn't support [single quotes in JSX](https://prettier.io/docs/en/options.html#quotes):

> Quotes in JSX will always be double and ignore this setting.

File: .eslintrc [hunk 1]
```diff
@@ -1,27 +1,20 @@
 {
   "parser": "babel-eslint",
-  "extends": "airbnb",
+  "extends": ["airbnb", "prettier"],
```
Comment (layershifter) [inline .eslintrc]: I think that we should stop use airbnb's preset

File: .eslintrc [hunk 2]
```diff
@@ -34,6 +27,7 @@
     "react/jsx-filename-extension": [2, { "extensions": [".js"] }],
     "react/no-unused-prop-types": 0,
     "react/sort-comp": 0,
-    "react/require-default-props": 0
+    "react/require-default-props": 0,
+    "prettier/prettier": "error"
```
Comment (layershifter) [inline .eslintrc]: Let's use numbers for rules config

File: .eslintrc (hunk 1 above)
Comment (thetric) [reply .eslintrc]: What are your reasons that you want to remove the `airbnb` preset?
I think that this is out of scope of the PR

**Comment by levithomason** [comment]
Let's keep this PR just adding the config.  We can test it locally.  This will help in reviewing.  We can merge the reformat later.

Also, other PRs can merge master, get the config update, and format their branches.

File: .eslintrc (hunk 1 above)
Comment (levithomason) [reply .eslintrc]: Prettier only covers stylistic issues, we'll still need eslint and a good preset for non-style issues such as unused vars.

**Comment by levithomason** [comment]
>Let's go with single quotes on JSX, it's a part of our codestyle and it will allow to reduce changes.

Yea, that sucks but it is what it is.  I pushed the point and got approval to submit a PR for the `singleQuote` option to respect JSX, however, jest was failing with cryptic errors and I gave up.  In the end, this is point of `prettier`.  Style issues just aren't worth the time.

**Comment by thetric** [comment]
> Let's keep this PR just adding the config. We can test it locally. This will help in reviewing. We can merge the reformat later.

Updated the PR to only add prettier and two config files

File: .prettierignore [hunk 3]
```diff
@@ -0,0 +1,16 @@
+.github/
```
Comment (levithomason) [inline .prettierignore:1]: _Deleted my comment, not applicable._

File: package.json [hunk 4]
```diff
@@ -28,6 +28,9 @@
     "deploy:docs": "gh-pages -d docs/build -m 'deploy docs [ci skip]'",
     "lint": "cross-env NODE_ENV=production eslint .",
     "lint:fix": "yarn lint --fix",
+    "prettier": "prettier --list-different \"**/*.{js,jsx,ts,tsx}\"",
+    "prettier:fix": "prettier --write \"**/*.{js,jsx,ts,tsx}\"",
```
Comment (thetric) [inline package.json]: Did you test it under Windows? In another project I had trouble with glob patterns as they tried to match files from `C:\` and not from the current directory. The fix was to prefix the glob pattern with `./`.

Comment (thetric) [inline package.json:32]: Did you test it under Windows? In another project I had trouble with glob patterns as they tried to match files from `C:\` and not from the current directory. The fix was to prefix the glob pattern with `./`.

Comment (levithomason) [reply package.json:32]: I did not.  Do you have a windows box for testing?  Perhaps @layershifter?

Note that I've escaped double quotes here, which means the globs will not be expanded by your shell (mac vs win) but rather by the node module that prettier uses under the hood, node-glob.  That module _should_ be platform agnostic, but I'm not 100% sure this is all correct.

Let me know how a Win test goes:

- edit a JS file, leaving a format mistake
- commit it (don't push)
- expect only that JS file to go through eslint and prettier
- repeat with a `ts` file, expecting tslint and prettier

Comment (thetric) [reply package.json:32]: Works on a Windows machine, too.

My trouble with the glob pattern under windows seems to be related to the quoting of the glob pattern. Using double quotes seems to fix it

**Comment by levithomason** [comment]
Are we good to merge here @layershifter and @thetric?

**Comment by thetric** [comment]
We could [upgrade prettier](https://prettier.io/blog/2018/04/11/1.12.0.html) before merging

**Comment by thetric** [comment]
I've nothing to add. What about you, @layershifter?

**Comment by layershifter** [comment]
@thetric Marked as ready to merge, great work :+1:

**Comment by levithomason** [comment]
Released in `semantic-ui-react@0.80.0`.

**Pull request merged**